src/m365/spo/commands/folder/folder-list.ts

```diff
@@ -55,26 +69,40 @@ class SpoFolderListCommand extends SpoCommand {
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     if (this.verbose) {
-      logger.logToStderr(`Retrieving folders from site ${args.options.webUrl} parent folder ${args.options.parentFolderUrl}...`);
+      logger.logToStderr(`Retrieving folders from site ${args.options.webUrl} parent folder ${args.options.parentFolderUrl}${args.options.recursive ? ' (recursive)' : ''}...`);
+    }
+
+    try {
+      const resp = await this.getFolderList(args.options.webUrl, args.options.parentFolderUrl, args.options.recursive);
+      logger.log(resp);
     }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
 
-    const serverRelativeUrl: string = urlUtil.getServerRelativePath(args.options.webUrl, args.options.parentFolderUrl);
-    const requestUrl: string = `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${encodeURIComponent(serverRelativeUrl)}')/folders`;
-    const requestOptions: any = {
+  private async getFolderList(webUrl: string, parentFolderUrl: string, recursive: boolean | undefined, folders: FolderProperties[] = []): Promise<FolderProperties[]> {
+    const serverRelativeUrl: string = urlUtil.getServerRelativePath(webUrl, parentFolderUrl);
+    const requestUrl: string = `${webUrl}/_api/web/GetFolderByServerRelativeUrl('${encodeURIComponent(serverRelativeUrl)}')/folders`;
+    const requestOptions: AxiosRequestConfig = {
       url: requestUrl,
       headers: {
         'accept': 'application/json;odata=nometadata'
       },
       responseType: 'json'
     };
 
-    try {
-      const resp = await request.get<{ value: FolderProperties[] }>(requestOptions);
-      logger.log(resp.value);
-    }
-    catch (err: any) {
-      this.handleRejectedODataJsonPromise(err);
+    const resp = await request.get<{ value: FolderProperties[] }>(requestOptions);
+    if (resp.value && resp.value.length > 0) {
```

Comment:
If I'm not mistaken, value can never be `undefined`, if nothing is found, an empty array will be returned.

```suggestion
    if (resp.value.length > 0) {
```